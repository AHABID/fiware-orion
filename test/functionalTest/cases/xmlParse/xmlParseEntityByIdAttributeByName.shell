echo "1: ++++++++++++++++++++"
# Make a registration using conv op /NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01
url="/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"
payload='<?xml version="1.0" encoding="UTF-8"?>
<registerProviderRequest>
  <metadata>
    <contextMetadata>
      <name>ID</name>
      <type>string</type>
      <value>1110</value>
    </contextMetadata>
    <contextMetadata>
      <name>cm2</name>
      <type>string</type>
      <value>XXX</value>
    </contextMetadata>
  </metadata>
  <duration>PT1H</duration>
  <providingApplication>http://kz.tid.es/abc</providingApplication>
</registerProviderRequest>'
curlXml ${url} "${payload}"

REG_ID=$(echo "$response" | grep registrationId | awk -F '>' '{print $2}' | awk -F '<' '{print $1}' | grep -v '^$' )

# Make a discovery to see that the registration went OK
echo "2: ++++++++++++++++++++"
url="/NGSI9/discoverContextAvailability"
payload='<?xml version="1.0"?>
  <discoverContextAvailabilityRequest>
    <entityIdList>
      <entityId type="" isPattern="true">
        <id>ENTITY_ID</id>
      </entityId>
    </entityIdList>
  </discoverContextAvailabilityRequest>'
curlXml ${url} "${payload}"

# Modify the first registration
echo "3: ++++++++++++++++++++"
url="/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"
payload='<?xml version="1.0" encoding="UTF-8"?>
<registerProviderRequest>
  <metadata>
    <contextMetadata>
      <name>ID</name>
      <type>string</type>
      <value>1111</value>
    </contextMetadata>
    <contextMetadata>
      <name>cm2</name>
      <type>string</type>
      <value>YYY</value>
    </contextMetadata>
  </metadata>
  <duration>PT2H</duration>
  <providingApplication>http://kz.tid.es/abc2</providingApplication>
  <registrationId>'$REG_ID'</registrationId>
</registerProviderRequest>'
curlXml ${url} "${payload}"

echo "4: ++++++++++++++++++++"

# Now discover that very registration again to see the change
url="/NGSI9/discoverContextAvailability"
payload='<?xml version="1.0"?>
  <discoverContextAvailabilityRequest>
    <entityIdList>
      <entityId type="" isPattern="true">
        <id>ENTITY_ID</id>
      </entityId>
    </entityIdList>
  </discoverContextAvailabilityRequest>'
curlXml ${url} "${payload}"

echo "5: ++++++++++++++++++++"

# Now try the GET method
curlXmlNoPayload "/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"

echo "6: ++++++++++++++++++++"

# And a GET that fails
curlXmlNoPayload "/NGSI9/contextEntities/ENTITY_ID2/attributes"

echo "7: ++++++++++++++++++++"

# DELETE and PUT should give errors
curlXmlNoPayload "/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01" "-X DELETE"

echo "8: ++++++++++++++++++++"

url="/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"
payload='<?xml version="1.0" encoding="UTF-8"?>
<registerProviderRequest>
  <metadata>
    <contextMetadata>
      <name>ID</name>
      <type>string</type>
      <value>1111</value>
    </contextMetadata>
    <contextMetadata>
      <name>cm2</name>
      <type>string</type>
      <value>YYY</value>
    </contextMetadata>
  </metadata>
  <duration>PT2H</duration>
  <providingApplication>http://kz.tid.es/abc2</providingApplication>
</registerProviderRequest>'
curlXml ${url} "${payload}" "-X PUT"

echo "9: ++++++++++++++++++++"

