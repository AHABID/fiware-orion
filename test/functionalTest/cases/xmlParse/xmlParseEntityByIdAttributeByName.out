1: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 141
Content-Type: application/xml
Date: Sun, 01 Jun 2014 17:45:34 GMT

<?xml version="1.0"?>
<registerContextResponse>
  <duration>PT1H</duration>
  <registrationId>538b66be0d4f0cdede01f069</registrationId>
</registerContextResponse>
2: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 770
Content-Type: application/xml
Date: Sun, 01 Jun 2014 17:45:34 GMT

<?xml version="1.0"?>
<discoverContextAvailabilityResponse>
  <contextRegistrationResponseList>
    <contextRegistrationResponse>
      <contextRegistration>
        <entityIdList>
          <entityId type="" isPattern="false">
            <id>ENTITY_ID</id>
          </entityId>
        </entityIdList>
        <contextRegistrationAttributeList>
          <contextRegistrationAttribute>
            <name>ATTRIBUTE_01</name>
            <type/>
            <isDomain>false</isDomain>
          </contextRegistrationAttribute>
        </contextRegistrationAttributeList>
        <providingApplication>http://kz.tid.es/abc</providingApplication>
      </contextRegistration>
    </contextRegistrationResponse>
  </contextRegistrationResponseList>
</discoverContextAvailabilityResponse>
3: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 141
Content-Type: application/xml
Date: Sun, 01 Jun 2014 17:45:34 GMT

<?xml version="1.0"?>
<registerContextResponse>
  <duration>PT2H</duration>
  <registrationId>538b66be0d4f0cdede01f069</registrationId>
</registerContextResponse>
4: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 771
Content-Type: application/xml
Date: Sun, 01 Jun 2014 17:45:34 GMT

<?xml version="1.0"?>
<discoverContextAvailabilityResponse>
  <contextRegistrationResponseList>
    <contextRegistrationResponse>
      <contextRegistration>
        <entityIdList>
          <entityId type="" isPattern="false">
            <id>ENTITY_ID</id>
          </entityId>
        </entityIdList>
        <contextRegistrationAttributeList>
          <contextRegistrationAttribute>
            <name>ATTRIBUTE_01</name>
            <type/>
            <isDomain>false</isDomain>
          </contextRegistrationAttribute>
        </contextRegistrationAttributeList>
        <providingApplication>http://kz.tid.es/abc2</providingApplication>
      </contextRegistration>
    </contextRegistrationResponse>
  </contextRegistrationResponseList>
</discoverContextAvailabilityResponse>
5: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 771
Content-Type: application/xml
Date: Sun, 01 Jun 2014 17:45:34 GMT

<?xml version="1.0"?>
<discoverContextAvailabilityResponse>
  <contextRegistrationResponseList>
    <contextRegistrationResponse>
      <contextRegistration>
        <entityIdList>
          <entityId type="" isPattern="false">
            <id>ENTITY_ID</id>
          </entityId>
        </entityIdList>
        <contextRegistrationAttributeList>
          <contextRegistrationAttribute>
            <name>ATTRIBUTE_01</name>
            <type/>
            <isDomain>false</isDomain>
          </contextRegistrationAttribute>
        </contextRegistrationAttributeList>
        <providingApplication>http://kz.tid.es/abc2</providingApplication>
      </contextRegistration>
    </contextRegistrationResponse>
  </contextRegistrationResponseList>
</discoverContextAvailabilityResponse>
6: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 185
Content-Type: application/xml
Date: Sun, 01 Jun 2014 17:45:34 GMT

<?xml version="1.0"?>
<discoverContextAvailabilityResponse>
  <errorCode>
    <code>404</code>
    <reasonPhrase>No context element found</reasonPhrase>
  </errorCode>
</discoverContextAvailabilityResponse>
7: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: Sun, 01 Jun 2014 17:45:34 GMT

8: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: Sun, 01 Jun 2014 17:45:34 GMT

9: ++++++++++++++++++++
